messaging/nats.go

```diff
@@ -43,14 +49,28 @@ func ConfigureNatsConnection(config *nconf.NatsConfig, log logrus.FieldLogger) (
 }
 
 func ConnectToNats(config *nconf.NatsConfig, opts ...nats.Option) (*nats.Conn, error) {
-	if config.TLS != nil {
-		tlsConfig, err := config.TLS.TLSConfig()
-		if err != nil {
-			return nil, errors.Wrap(err, "Failed to configure TLS")
-		}
-		if tlsConfig != nil {
-			opts = append(opts, nats.Secure(tlsConfig))
+	tlsConfig, err := config.TLS.TLSConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to configure TLS config")
+	}
+
+	// If TLS is enabled, always add the Secure option: it will at least contain the RootCAs
+	if config.TLS.Enabled {
+		opts = append(opts, nats.Secure(tlsConfig))
+	}
+
+	switch strings.ToLower(config.Auth.Method) {
+	case nconf.NatsAuthMethodUser:
+		opts = append(opts, nats.UserInfo(config.Auth.User, config.Auth.Password))
+	case nconf.NatsAuthMethodToken:
+		opts = append(opts, nats.Token(config.Auth.Token))
+	case nconf.NatsAuthMethodTLS:
+		// if using TLS auth, make sure the client certificate is loaded
```

Comment:
I think that doing the TLS stuff in here works well 

```
case nconf.NatsAuthMethodTLS:
  if !config.TLS.Enabled { return an error }
  
  tlsConfig, err := config.TLS.TLSConfig()
  if err != nil {
    return nil, errors.Wrap(err, "Failed to configure TLS config")
  }
  if len(tlsConfig.Certificates) == 0 {
    return nil, fmt.Errorf("TLS auth method is configured but no certificate was loaded")
  }
  
  opts = append(opts, nats.Secure(tlsConfig))
default: 
...
```

Reply:
we want the TLS conf always, even if the auth is not TLS, because we still need encrypted connections to the servers.

The only difference is for `tls` auth, we also need the certificate. For the rest, we only need the RootCAs

Reply:
ah good point I didn't see that. If that is the case then I'd probably want to be more careful around:
```
if len(tlsConfig.Certificates) == 0
```

`tlsConfig` could be nil I think (shouldn't be) but it would cause a panic instead of an error message